app/grandchallenge/algorithms/views.py

```diff
@@ -335,6 +338,82 @@ def get_context_data(self, *args, **kwargs):
         return context
 
 
+class AlgorithmImageMarkDesired(
+    LoginRequiredMixin, ObjectPermissionRequiredMixin, View
+):
+    permission_required = "algorithms.change_algorithm"
+    raise_exception = True
+
+    @cached_property
+    def algorithm(self) -> Algorithm:
+        return self.algorithm_image.algorithm
+
+    @cached_property
+    def algorithm_image(self) -> AlgorithmImage:
+        return get_object_or_404(AlgorithmImage, pk=self.kwargs["pk"])
+
+    def get_permission_object(self):
+        return self.algorithm
+
+    def post(self, request, *args, **kwargs):
+        if not self.algorithm_image.is_manifest_valid:
+            messages.error(
+                request, "Cannot mark an invalid image as active image."
+            )
+            return HttpResponseRedirect(
+                self.algorithm_image.get_absolute_url()
+            )
+
+        if not self.algorithm_image.can_execute:
+            image_upload_in_progress = (
+                self.algorithm_image.import_status
+                == ImportStatusChoices.STARTED
+            )
+            if not image_upload_in_progress:
+                self.algorithm_image.import_status = (
+                    ImportStatusChoices.STARTED
+                )
+                self.algorithm_image.save()
```

Comment:
This should not be set here, this should only be set when a worker claims the job. You may need to add another attribute to the container image model to track this, or maybe it should be set to pending and you check for

```python
            image_upload_in_progress = (
                self.algorithm_image.import_status
                in (ImportStatusChoices.STARTED, ImportStatusChoices.PENDING)
            )
```

Additionally, `image_upload_in_progress` needs to check ALL images for the related algorithm to prevent contesting imports, so something like:

```python
@property
def image_upload_in_progress(self):
    return self.algorithm_image_set.filter(status__in=(ImportStatusChoices.STARTED, ImportStatusChoices.PENDING)).exists()
```

on the algorithm model.

Reply:
Good point about checking for contesting imports! I think the "pending" (queued) status works well, but I will still need to set this here on the view - otherwise, calling the view a second time when the task hasn't been picked up by a worker yet would result in another task being scheduled.